tutorial/managing-software.rst

```diff
@@ -0,0 +1,1611 @@
+
+Instead of **versions** as we have them in debs, snaps use the concept of
+`channels <https://snapcraft.io/docs/channels>`_ to define which release of a
+snap is installed.
+
+By default, snaps are kept automatically up-to-date, so we don't need to
+remember to update and upgrade them. There are times on a live system, such as
+a server in a production environment, where we might not want to have updates
+automatically applied. In those cases, we can
+`turn off automatic updates <https://snapcraft.io/docs/managing-updates>`_ and
+refresh the system snaps when it's convenient (for example, during a
+maintenance window).
+
+If you would like to try out snaps, we recommend the excellent
+`quickstart tour <https://snapcraft.io/docs/quickstart-tour>`_ tutorial in the
+snap documentation. Feel free to continue using the VM we've been using in
+this tutorial while exploring!
+
+Completion!
+===========
+
+Once you are finished and want to leave the tutorial, you can run:
+
+.. code-block:: bash
+
+    exit
+
+This will take you out of the VM and back to your live machine. Then, you can
+run the following commands to delete the VM and remove it completely from your
+machine:
+
+.. code-block:: bash
+
+    multipass delete tutorial
+    multipass purge
+
+Summary
+=======
+
+Congratulations, we made it to the end! We've covered a lot of material in this
+tutorial, so let's do a quick recap of what we've learned:
+
+**Finding, installing and removing packages**
+
+* How to update and upgrade all our system's software with APT:
+  * ``sudo apt update && sudo apt upgrade``
+
+* How to search for software using keywords or strings:
+  * ``apt search <keyword>`` or ``apt search "some content string"`` 
+
+* How to see the description of a package, including what dependencies it has:
+  * ``apt show <package name>``
+  
+  Or how to check what package versions are available:
+  * ``apt policy <package>``
+
+* How to install packages...
+  * ``sudo apt install <package1> <package2>``
+
+* How to see all the files a package contains
+  * ``dpkg --listfiles <package>``
+
+* How to find out what package a file belongs to:
+  * ``dpkg --search <path/to/file>``
+
+* ...And how to remove packages again! As well as the difference between removing and purging.
+  * ``sudo apt remove <package>``
+
+* We even learned how to downgrade to older versions of APT packages, and all
+  about APT sources.
+
+**Customising packages**
+
+* How to find the conffiles in a package:
+  * ``dpkg-query --show -f='${Conffiles}\n' <package>``
+
+* How to see if package files have been changed:
+  * ``dpkg --verify <package>``
+
+* ...And if a non-conffile has been changed by accident, we can fix it with:
+  * ``sudo apt install --reinstall <package>``
+
+* We know that our changes to conffiles are always safely preserved, while
+  changes to non-conffiles are reverted at the next upgrade or security fix.
+
+* Importantly, we know how to verify checksums with ``md5sum`` or similar
```

Comment:
You did not mention the build from source here, which is good - I do not want it here :-)

But this list also forgets about the install from a .deb file - which ok'ish.
I'm unsure if that should be here or also in a page of its own with big disclaimers (essentially the third party explanation we have).

Reply:
I don't want to draw too much attention to it, since we prefer people don't do it and are only showing it as an FYI because we're doing it in a sandbox. Actually, now that I reflect more on the reasons to remove all the building from source stuff, I wonder if we should also remove the install from .deb stuff - but it does provide a convenient segue into snaps.